### PR TITLE
Assert Android API >= v21

### PR DESCRIPTION
**Pull request by drfloob**
This precedes a change that would otherwise break the same Android APIs, but this PR breaks the build in a more obvious way. Android SDK v20 corresponds to KitKat, which was released in 2014. gRPC is dropping support for older android SDKs, which may affect a small chunk of circulating Android devices today.

Why drop this support? We are working to integrate libuv into gRPC core, and libuv does not support Android APIs before v21.

@veblush

### CODE REVIEW

**Comment by drfloob** [comment]
Superseded by https://github.com/grpc/grpc/pull/27912.

**Comment by sifmelcara** [comment]
I think this is not Superseded by #27912

#27912 changes `29` in the following code to `23`
```
#if (__ANDROID_API__) >= 29
#define GPR_SUPPORT_BINDER_TRANSPORT 1
#endif
```
gRPC will still compile when `GPR_SUPPORT_BINDER_TRANSPORT` is not defined. If it is not defined, we just disables some specific binder transport related APIs

**Comment by drfloob** [comment]
Ooh, my mistake! Thanks for letting me know.

**Comment by sifmelcara** [comment]
The `Android (Internal CI)` kokoro build fails; seems like it passes `-D__ANDROID_API__=14` to the compiler?

File: include/grpc/impl/codegen/port_platform.h [hunk 1]
```diff
@@ -127,6 +127,9 @@
 #define GPR_PLATFORM_STRING "android"
 #define GPR_ANDROID 1
 #ifdef __ANDROID_API__
```
Comment (sifmelcara) [inline include/grpc/impl/codegen/port_platform.h:129]: Although `__ANDROID_API__` should always be defined, we probably want to also emit an error if it is not defined, just to be safe

**Review by sifmelcara — APPROVED** [review]
Looks good if manual copybara import passes

**Review by veblush — APPROVED** [review]
Thanks for doing this!

**Comment by drfloob** [comment]
PR check failures are: 2 timeouts and one infrastructure failure (details page is missing entirely). Unrelated to this change.

**Pull request merged**